agent/build.gradle.kts

```diff
@@ -26,104 +26,27 @@ dependencies {
-      include("NOTICE")
-      include("licenses/**")
     }
-
   }
 
   assemble {
     dependsOn(javadocJar, sourcesJar)
   }
-
-  val licensesDir = rootDir.toPath().resolve("licenses")
-  val licenseFile = "more-licences.md"
-
-  licenseReport {
-    outputDir = licensesDir.toString()
-    renderers = arrayOf(InventoryMarkdownReportRenderer(licenseFile))
-    excludeBoms = true
-    excludes = arrayOf(
-      "io.opentelemetry:opentelemetry-bom-alpha",
-      "io.opentelemetry.instrumentation:opentelemetry-instrumentation-bom-alpha"
-    )
-    filters = arrayOf(
-      LicenseBundleNormalizer(
-        "${rootProject.rootDir}/buildscripts/license-normalizer-bundle.json",
-        true
-      )
-    )
-    projects = arrayOf(
-      rootProject.project("agent"), rootProject.project("bootstrap"),
-      rootProject.project("common"), rootProject.project("custom"),
-      rootProject.project("inferred-spans"), rootProject.project("instrumentation"),
-      rootProject.project("resources")
-    )
-
-    configurations = arrayOf("runtimeClasspath", "compileClasspath")
-  }
-
-  register("updateLicensesAndNotice") {
-    dependsOn(generateLicenseReport)
-    doLast {
-
-      var year = Calendar.getInstance().get(Calendar.YEAR)
-      var lines = ArrayList<String>(
-        listOf(
-          "Elastic Distribution of OpenTelemetry Java",
-          "Copyright 2023-${year} Elasticsearch B.V.",
-          "",
-          "This project is licensed under the Apache License, Version 2.0 - https://www.apache.org/licenses/LICENSE-2.0",
-          "A copy of the Apache License, Version 2.0 is provided in the 'LICENSE' file.",
-          "",
-          "This project depends on ASM which is licensed under the 3-Clause BSD License - https://opensource.org/licenses/BSD-3-Clause",
-          "A copy of the ASM license (3-Clause BSD License) is provided in the 'licenses/LICENSE_asm_bsd-3-clause' file.",
-          "",
-          "This project depends on okhttp, which contains 'publicsuffixes.gz' that is licensed under the Mozilla Public License, v. 2.0 - https://mozilla.org/MPL/2.0/",
-          "A copy of the publicsuffixes.gz license (Mozilla Public License, v. 2.0) is provided in the 'licenses/LICENSE_mpl-2' file.",
-          "",
-          "This project depends on HdrHistogram which is dual-licensed under the following licenses:",
-          "  - 2-Clause BSD License - https://opensource.org/licenses/BSD-2-Clause",
-          "  - Creative Commons Public Domain License - https://creativecommons.org/publicdomain/zero/1.0/",
-          "A copy of the HdrHistogram license is provided in the 'licenses/LICENSE_hdrhistogram_bsd-2-clause_cc0' file.",
-          "",
-          "Details on the individual dependencies notices and licenses can be found in the 'licenses' folder.",
-          ""
-        )
-      )
-
-      Files.write(rootDir.toPath().resolve("NOTICE"), lines)
-
-      // make the generated license report idempotent by removing the date
```

Comment:
This part is still needed otherwise it makes the build non-idempotent due to the date generated in `more-licenses.md`.

Reply:
Done in https://github.com/elastic/elastic-otel-java/pull/409/commits/acda4aced88c62fab8629391ad53d9638283616a